layers/shader_object/shader_object.cpp

```diff
@@ -630,6 +630,157 @@ static void RemoveCommandBufferDataForCommandBuffer(DeviceData* device_data, VkC
     }
 }
 
+static std::string GetShaderName(uint32_t shader_type) {
+    switch (shader_type) {
+        case VERTEX_SHADER:
+            return "V";
+        case FRAGMENT_SHADER:
+            return "F";
+        case TESSELLATION_CONTROL_SHADER:
+            return "TC";
+        case TESSELLATION_EVALUATION_SHADER:
+            return "TE";
+        case GEOMETRY_SHADER:
+            return "G";
+        case MESH_SHADER:
+            return "M";
+        case TASK_SHADER:
+            return "T";
+        default:
+            break;
+    }
+    return "";
+}
+
+static void SetComputeShaderDebugUtilsName(DeviceData& data, Shader* shader, const VkDebugUtilsObjectNameInfoEXT *pNameInfo) {
+    VkDebugUtilsObjectNameInfoEXT name_info{
+        VK_STRUCTURE_TYPE_DEBUG_UTILS_OBJECT_NAME_INFO_EXT,
+        nullptr,
+        VK_OBJECT_TYPE_PIPELINE,
+        (uint64_t)shader->partial_pipeline.pipeline,
+        pNameInfo->pObjectName
+    };
+    data.vtable.SetDebugUtilsObjectNameEXT(data.device, &name_info);
+}
+
+static void SetComputeShaderDebugUtilsTag(DeviceData& data, Shader* shader, const VkDebugUtilsObjectTagInfoEXT* pTagInfo) {
+    VkDebugUtilsObjectTagInfoEXT tag_info{
+        VK_STRUCTURE_TYPE_DEBUG_UTILS_OBJECT_TAG_INFO_EXT,
+        nullptr,
+        VK_OBJECT_TYPE_PIPELINE,
+        (uint64_t)shader->partial_pipeline.pipeline,
+        pTagInfo->tagName,
+        pTagInfo->tagSize,
+        pTagInfo->pTag
+    };
+    data.vtable.SetDebugUtilsObjectTagEXT(data.device, &tag_info);
+}
+
+static void SetDebugUtilsNameAndTag(CommandBufferData& cmd_data, VkPipeline pipeline) {
+    auto& device_data = *cmd_data.device_data;
+    auto const state  = cmd_data.GetDrawStateData();
+
+    if (device_data.debug_utils_object_name_map.NumEntries() > 0) {
+        bool first_shader = true;
+        bool same_name = true;
+        char same_shader_name[128];
+        char pipeline_name[256];
+        pipeline_name[0] = '\0';
+        char temp[128];
+
+        for (uint32_t shader_type = 0; shader_type < NUM_SHADERS; ++shader_type) {
+            Shader* shader = state->GetComparableShader(shader_type).GetShaderPtr();
+
+            if (shader == nullptr) {
+                continue;
+            }
+
+            auto iter = device_data.debug_utils_object_name_map.Find(shader);
+            if (iter != device_data.debug_utils_object_name_map.end()) {
+                const auto& shader_name = iter.GetValue();
+                if (first_shader) {
+                    first_shader = false;
+                    strcpy(same_shader_name, shader_name.c_str());
```

Comment:
There's 2 choices here: 1.  use (strncat, strncpy, snprintf) 2. use std::stringstream.  I'd prefer the latter if possible. This isn't a performance critical path.

Reply:
It is a performance critical path, since you get here from a `vkCmdDraw()`